var/spack/repos/builtin/packages/git/package.py

```diff
@@ -174,6 +177,7 @@ class Git(AutotoolsPackage):
     depends_on('automake', type='build')
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
+    depends_on('tk',       type='run', when='+tcltk')
```

Comment:
This *might* be a build dependency it we pass the correct path at run-time to `--with-tcltk`. I understand your point on `tk` not being strictly needed during compilation, but maybe it's worth considering:
1. Making `tk` a `(build, link)` dependency conditional on the variant
2. Pass its prefix to the configure script if `+tcltk`